go-controller/pkg/ovn/egressgw.go

```diff
@@ -467,8 +472,10 @@ func (oc *Controller) addGWRoutesForPod(gateways []*gatewayInfo, podIfAddrs []*n
 					if err != nil && !strings.Contains(stderr, DuplicateECMPError) {
 						return fmt.Errorf("unable to add external gwStr src-ip route to GR router, stderr:%q, err:%gw", stderr, err)
 					}
-					if err := oc.addHybridRoutePolicyForPod(podIPNet.IP, node); err != nil {
-						return err
+					if len(routeInfo.podExternalRoutes[podIP]) == 0 {
```

Comment:
@trozet is this supposed to be `> 0` ?

Reply:
No because that would make it configure hybrid policy for every external route loop. We only need it to do it 1 time, so at this point in the code we know we have added a route into OVN, but have not incremented the cache yet so I check for 0 (its about to be 1 after this check). I guess this could be more obvious if it was moved under line 482 and set to 
```if len(routeInfo.podExternalRoutes[podIP]) == 1 {
```

wdyt?

Reply:
@trozet I get it now. Sounds good.